### PR TITLE
Add format/noreturn attributes and fix some warnings

### PR DESCRIPTION
**Pull request by dnk777**

### CODE REVIEW

File: source/game/ai/bot_tactical_spots_cache.cpp [hunk 1]
```diff
@@ -330,7 +330,7 @@ void BotTacticalSpotsCache::FindReachableClassEntities( const Vec3 &origin, floa
 	AiAasRouteCache *routeCache = self->ai->botRef->routeCache;
 
 	bool testTwoCurrAreas = false;
-	int fromAreaNum;
+	int fromAreaNum = 0;
```
Comment (viciious) [inline source/game/ai/bot_tactical_spots_cache.cpp:333]: Should probably default to -1

Comment (dnk777) [reply source/game/ai/bot_tactical_spots_cache.cpp:333]: Zero is an illegal number for an area (used for a dummy area by AAS conventions). Negative area numbers are never (should be never) met in the code.

**Comment by dnk777** [comment]
Can you tell did the memset call suppress the corresponding warning?

**Comment by viciious** [comment]
It obviously does

**Comment by dnk777** [comment]
Wait please I want to add something else...

**Comment by viciious** [comment]
Android build fails: https://travis-ci.org/Qfusion/qfusion/jobs/249105175

**Comment by dnk777** [comment]
What a can of warms has been opened https://gist.github.com/dnk777/25488002c38d771003bd060491cf2d19

**Comment by dnk777** [comment]
Btw the only proper way to print an int64_t is using the PRId64 macro (its ugly I know).

**Comment by viciious** [comment]
Yes, but I'm not sure whether these macros defined for MSVC..

**Comment by dnk777** [comment]
I used this macro in the GOAP branch and it has passed appveyour builds.

**Comment by dnk777** [comment]
Looks like MSVC does not support `__declspec(noreturn)` attribute  application to a function pointer variable in all import structures.

**Comment by dnk777** [comment]
Looks like IRC stuff has not been removed completely from cmake files.
```
CMake Warning (dev) at server/CMakeLists.txt:90 (add_dependencies):
  Policy CMP0046 is not set: Error on non-existent dependency in
  add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.
  The dependency target "irc" of target "qfusion_server" does not exist.
This warning is for project developers.  Use -Wno-dev to suppress it.
CMake Warning (dev) at client/CMakeLists.txt:198 (add_dependencies):
  Policy CMP0046 is not set: Error on non-existent dependency in
  add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.
  The dependency target "irc" of target "qfusion" does not exist.
This warning is for project developers.  Use -Wno-dev to suppress it.
-- Generating done
-- Build files have been written to: C:/projects/qfusion/build_x86
```

I am not sure whether other subprojects have IRC dependency too

**Pull request merged**